bounding-boxes/src/coco_to_datasaur_schemas.py

```diff
@@ -0,0 +1,264 @@
+        default="./outdir/",
+    )
+    parser.add_argument("--log-level", type=str, default="INFO")
+    args = parser.parse_args()
+    logging.basicConfig(level=args.log_level, format="%(message)s")
+
+    coco_filepath = os.path.abspath(args.coco_filepath)
+    log("reading COCO JSON file", filepath=coco_filepath)
+    with open(coco_filepath) as f:
+        json_data = json.load(f)
+
+    custom_labelset = None
+    if args.custom_labelset:
+        custom_labelset_filepath = os.path.abspath(args.custom_labelset)
+        with open(custom_labelset_filepath) as f:
+            custom_labelset = json.load(f)
+
+    log("converting COCO to Datasaur Schema")
+    schemas = coco_to_datasaur_schemas(json_data, custom_labelset)
+
+    outdir = os.path.abspath(args.outdir)
+    os.makedirs(outdir, exist_ok=True)
+    log("writing to file", directory=outdir)
+    for schema in schemas:
+        filename = schema["data"]["document"]["name"].split(".")[0] + ".json"
+
+        with open(os.path.join(outdir, filename), "w") as f:
+            json.dump(scrub(schema), f, indent=2)
+
+
+def bbox_label_classes_from_coco(
+    coco_categories: List[dict],
+    custom_labelset: Any | None,
+) -> List[DSBBoxLabelClass]:
+    def defaults(obj: dict | None, key: str, default: Any):
+        return obj[key] if obj and key in obj else default
+
+    custom_classes = custom_labelset["classes"] if custom_labelset else []
+
+    retval = []
+    for category in coco_categories:
+        # check and use questions from custom class
+        custom_class = next(
+            (item for item in custom_classes if item["name"] == category["name"]), None
+        )
+
+        questions = [
+            DSBBoxLabelClassQuestions(
+                id=q.get("id", index),
+                label=q.get("label", f"Question {index}"),
+                config=QuestionConfig(
+                    multiline=q.get("config", {}).get("multiline", False),
+                    multiple=q.get("config", {}).get("multiple", False),
+                    options=q.get("config", {}).get("options", []),
+                    defaultValue=q.get("config", {}).get("defaultValue", None),
+                ),
+                required=q.get("required", False),
+                type=q.get("type", "TEXT"),
+            )
+            for index, q in enumerate(defaults(custom_class, "questions", []))
+        ]
+
+        retval.append(
+            DSBBoxLabelClass(
+                id=str(category["id"]),
+                name=category["name"],
+                captionAllowed=defaults(custom_class, "captionAllowed", True),
+                captionRequired=defaults(custom_class, "captionRequired", False),
+                color=defaults(custom_class, "color", random_color(category["name"])),
+                questions=questions,
+            )
+        )
+
+    return retval
+
+
+def shape_from_coco_segmentation(segmentation: List[float]) -> DSShape:
+    if len(segmentation) != 8:
```

Comment:
8 nya dijadiin variabel sendiri mungkin? karena aku liat dipake di beberapa tempat

atau kondisi ini jadi function sendiri? karena di atas juga ada pengecekan yg sama

Reply:
ho iya
benernya validasi disini udah ga perlu sih (kalau dipanggil as a script), karena diprevent didepan. 


kemarin jaga-jaga kalau dipakai as a module dan fungsi ini dipake langsung. will do!

Reply:
moved to `formats/coco`